source/gui/__init__.py

```diff
@@ -220,7 +220,9 @@ def onSaveConfigurationCommand(self,evt):
 	def popupSettingsDialog(self, dialog: Type[SettingsDialog], *args, **kwargs):
 		self.prePopup()
 		try:
-			dialog(self, *args, **kwargs).Show()
+			d = dialog(self, *args, **kwargs)
+			if d:
+				d.Show()

```

Comment:
should there be some error logging if d is not created?

```suggestion
				d.Show()
			else:
				log.error(...)
```

Reply:
Additional logging here would be redundant, as the only case in which `__new__` returned `None` is when there were multiple instances, which is logged already. Having said that I have simplified the code, so this is no longer a concern.